projects/pyjwt/fuzz_jwt.py

```diff
@@ -18,29 +18,67 @@
 import atheris
 
 import jwt
+from jwt.algorithms import RSAAlgorithm
+from jwt.api_jws import PyJWS
 import json
 from collections.abc import Mapping
 
 
+def ConsumeRandomLengthString(fdp, data):
+    """Creates a random string of random length between 0 - 1024"""
+    fdp = atheris.FuzzedDataProvider(data)
+    b = fdp.ConsumeUnicodeNoSurrogates(fdp.ConsumeIntInRange(0, 1024))
+    return b
+
+
 def test_decoding(data):
     """Checks jwt decoding does not fail"""
-    fdp = atheris.FuzzedDataProvider(data)
-    s1 = fdp.ConsumeUnicodeNoSurrogates(1024)
+    s1 = data
     try:
         _ = jwt.decode(s1, algorithms=["HS256"])
     except jwt.exceptions.PyJWTError:
         pass
+    try:
+        _ = jwt.api_jwt.decode_complete(jwt=s1, algorithms=["HS256"])
+    except jwt.exceptions.PyJWTError:
+        pass
+    
+
+def test_JWS_encoding(data):
+    """Checks PyJWS encoding does not fail"""
+    payload = data
+    key = "fuzzing"
+    PyJWS().encode(payload=payload, key=key)
+
 
+def test_roundtrip_with_RS256(data):
+    """Check payload == decoded(encoded(payload)) using RS256"""
+    try:
+        payload = json.loads(data)
+        keyfile = json.loads(data)
```

Comment:
Let's make `payload` and `keyfile` different (i.e different bytes for `json.loads`).